cmd/rancherd/auth/auth.go

```diff
@@ -142,6 +146,9 @@ func ResetAdmin(_ *cli.Context) error {
 			return err
 		}
 		adminName = admin.GetName()
+		if !done {
```

Comment:
It doesn't need to be called twice and return a bool. It can be called at the end of the reset-admin method.
```
func setClusterAnnotation() {
  if annotation is not there:
    setAnnotation();
}
```